light/provider/http/http.go

```diff
@@ -238,11 +238,17 @@ func (p *http) validatorSet(ctx context.Context, height *int64) (*types.Validato
 			break
 		}
 	}
+	valSet := types.NewValidatorSet(vals, thresholdPubKey, quorumType, quorumHash, false)
 
-	valSet, err := types.ValidatorSetFromExistingValidators(vals, thresholdPubKey, quorumType, quorumHash)
-	if err != nil {
-		return nil, provider.ErrBadLightBlock{Reason: err}
+	if valSet == nil || valSet.IsNilOrEmpty() {
+		return nil, provider.ErrBadLightBlock{Reason: fmt.Errorf("retrieved nil or empty validator set")}
+	}
+	if err := valSet.ValidateBasic(); err != nil {
+		return nil, provider.ErrBadLightBlock{Reason: fmt.Errorf("invalid validator set retrieved: %w", err)}
 	}
+
+	valSet.SetProposer(proposer)
```

Comment:
linting

Reply:
Error return value of `valSet.SetProposer` is not checked (errcheck)

Reply:
fixed